File: TDS/TDS/Models/Estudiante.cs
using System;
using System.Collections.Generic;

namespace TDS.Models
{
    public partial class Estudiante
    {
        public Estudiante()
        {
            Entregas = new HashSet<Entrega>();
            EstudiantesClases = new HashSet<EstudiantesClase>();
            Mensajes = new HashSet<Mensaje>();
            MensajesDetalles = new HashSet<MensajesDetalle>();
            Usuarios = new HashSet<Usuario>();
        }

        public int Id { get; set; }
        public string? Nombres { get; set; }
        public string? Apellidos { get; set; }
        public string? Correo { get; set; }
        public string? Codigo { get; set; }
        public string? Direccion { get; set; }
        public string? Telefono { get; set; }
        public bool? Estado { get; set; }
        public int? InstitucionId { get; set; }

        public virtual Institucion? Institucion { get; set; }
        public virtual ICollection<Entrega> Entregas { get; set; }
        public virtual ICollection<EstudiantesClase> EstudiantesClases { get; set; }
        public virtual ICollection<Mensaje> Mensajes { get; set; }
        public virtual ICollection<MensajesDetalle> MensajesDetalles { get; set; }
        public virtual ICollection<Usuario> Usuarios { get; set; }
    }
}

File: TDS_Front/src/app/models/estudiante.ts
import { Entrega } from "./entrega";
import { EstudiantesClases } from "./estudiantes-clases";

export class Estudiante {
  constructor(public id: number = 0, public nombres: string = "", public apellidos: string = "", public correo: string = "", public codigo: string = "", public direccion: string = "", public telefono: string = "", public estado: boolean = false, public institucionId: number = 0, public estudiantesClases: EstudiantesClases[] = [], public entregas: Entrega[] = []){}
}
